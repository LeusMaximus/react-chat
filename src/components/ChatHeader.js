// React
import React from 'react';

// MUI components
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

// Own modules
import Avatar from './Avatar';
import UserNav from '../containers/UserNav';

const styles = theme => ({
  appBar: {
    flexShrink: 1,
    minHeight: 64,
  },

  accountBox: {
    marginLeft: 'auto',
  },

  chatAvatar: {
    marginRight: '10px',
  },
});

const ChatHeader = ({ classes, chatName }) => (
  <AppBar
    position="static"
    className={classes.appBar}
  >
    <Toolbar>
      <Avatar className={classes.chatAvatar}>
        {chatName}
      </Avatar>

      <Typography variant="title" color="inherit" noWrap>
        {chatName}
      </Typography>

      <UserNav className={classes.accountBox} />
    </Toolbar>
  </AppBar>
);

export default withStyles(styles)(ChatHeader);
