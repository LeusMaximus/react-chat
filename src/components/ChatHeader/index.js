// React
import React from 'react';
import PropTypes from 'prop-types';

// MUI components
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

// Vendor modules
import isEmpty from 'lodash/isEmpty';
import _get from 'lodash/get';

// Own modules
import Avatar from '../../components/Avatar';
import UserNav from '../../containers/UserNav';
import ChatMenu from '../../components/ChatMenu';
import { IClasses, IChatItem } from '../../interfaces/propTypes';

const styles = () => ({
  appBar: {
    flexShrink: 1,
    minHeight: 64,
  },

  accountBox: {
    marginLeft: 'auto',
  },

  chatAvatar: {
    marginRight: 10,
  },

  chatMenu: {
    marginRight: 10,
  },
});

const ChatHeader = ({
  classes,
  activeChat,
  activeId,
  isMember,
  isCreator,
  isChatMember,
  leaveChat,
  deleteChat,
  isConnected,
}) => (
  <AppBar position="static" className={classes.appBar}>
    <Toolbar>
      {!isEmpty(activeChat) &&
        isChatMember && (
          <ChatMenu
            disabled={!isConnected}
            className={classes.chatMenu}
            isMember={isMember}
            isCreator={isCreator}
            leaveChat={leaveChat}
            deleteChat={deleteChat}
            activeId={activeId}
          />
        )}

      {!isEmpty(activeChat) && <Avatar className={classes.chatAvatar}>{activeChat.title}</Avatar>}

      <Typography variant="title" color="inherit" noWrap>
        {_get(activeChat, 'title') || 'Super Messenger'}
      </Typography>

      <UserNav className={classes.accountBox} disabled={!isConnected} />
    </Toolbar>
  </AppBar>
);

ChatHeader.defaultProps = {
  classes: null,
  activeChat: {},
};

ChatHeader.propTypes = {
  classes: IClasses,
  activeChat: IChatItem,
  activeId: PropTypes.string.isRequired,
  isMember: PropTypes.bool.isRequired,
  isCreator: PropTypes.bool.isRequired,
  isChatMember: PropTypes.bool.isRequired,
  leaveChat: PropTypes.func.isRequired,
  deleteChat: PropTypes.func.isRequired,
  isConnected: PropTypes.bool.isRequired,
};

export default withStyles(styles)(ChatHeader);
