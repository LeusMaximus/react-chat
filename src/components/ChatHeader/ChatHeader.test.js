/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import ChatHeader from './index';

jest.mock('../ChatMenu', () => () => 'ChatMenu');
jest.mock('../Avatar', () => () => 'Avatar');
jest.mock('../../containers/UserNav', () => () => 'UserNav');

const mockProps = {
  activeChat: {
    _id: '1000',
    updatedAt: '2017-04-14T11:45:35.200Z',
    createdAt: '2017-03-10T10:35:25.200Z',
    title: 'Test Chat',
  },
  activeId: '1000',
  isMember: false,
  isCreator: true,
  isChatMember: true,
  leaveChat: jest.fn(),
  deleteChat: jest.fn(),
  isConnected: true,
};

describe('<ChatHeader />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatHeader {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot test when active chat is', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('snapshot test when no any active chats', () => {
    const tree = renderer.create(<ChatHeader {...mockProps} activeChat={null} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
