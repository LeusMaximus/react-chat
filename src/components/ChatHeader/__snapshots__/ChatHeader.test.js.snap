// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> snapshot test when active chat is 1`] = `
<header
  className="MuiPaper-root-78 MuiPaper-elevation4-84 MuiAppBar-root-70 MuiAppBar-positionStatic-74 MuiAppBar-colorPrimary-76 ChatHeader-appBar-66"
>
  <div
    className="MuiToolbar-root-105 MuiToolbar-gutters-106"
  >
    ChatMenu
    Avatar
    <h2
      className="MuiTypography-root-107 MuiTypography-title-113 MuiTypography-colorInherit-126 MuiTypography-noWrap-123"
    >
      Test Chat
    </h2>
    UserNav
  </div>
</header>
`;

exports[`<ChatHeader /> snapshot test when no any active chats 1`] = `
<header
  className="MuiPaper-root-78 MuiPaper-elevation4-84 MuiAppBar-root-70 MuiAppBar-positionStatic-74 MuiAppBar-colorPrimary-76 ChatHeader-appBar-66"
>
  <div
    className="MuiToolbar-root-105 MuiToolbar-gutters-106"
  >
    <h2
      className="MuiTypography-root-107 MuiTypography-title-113 MuiTypography-colorInherit-126 MuiTypography-noWrap-123"
    >
      Super Messenger
    </h2>
    UserNav
  </div>
</header>
`;
